### ISSUE TITLE
add scripts to collect stats for the repo

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#3791]
in view of the OHBM hackathon townhall (some info may be usefull for grant writing?), try to automate the following:
- new contributors over a given period of time
- open / close issues over a period of time
- open / close PRs
- time to fix for bugs
- average time for a PR (to help give contributors an estimate how long the process is)

This will probably require a lot of calls to the github APIs

**Comment by emdupre** [comment]
I wonder if we could re-use some of the structure from this live tracker : https://github.com/abbycabs/hackathon-live-feed

**Comment by Remi-Gau** [comment]
Thing to add: number of repos and package that depend on nilearn

https://github.com/nilearn/nilearn/network/dependents?dependent_type=REPOSITORY

**Comment by FrancoisPgm** [comment]
Hi, I'm down to give it a try :)

**Comment by FrancoisPgm** [comment]
From a quick google search I only found apps to have a dashboard on your local machine, and for the hackaton-live-feed it seems that you need to run it on a server, so I'm not sure where it would be deployed. But I found [a tutorial](https://medium.com/swlh/how-to-create-a-self-updating-readme-md-for-your-github-profile-f8b05744ca91) to make a dynamic readme that updates regularly through github actions. I can try to make something similar for a markdown file that would live in /doc or at the root, not necessaraly the README directly.

**Comment by Remi-Gau** [comment]
let's chat this afternoon so I can show you some of the dummy scripts I wrote when I played with this

**Comment by bthirion** [comment]
Interested in the result of the discussion. We should also talk to the sklearn people, they monitor well their activity.

**Comment by man-shu** [comment]
Was there any progress on this? If not, it's still a good Hackathon issue.

**Comment by Remi-Gau** [comment]
I know that @FrancoisPgm had worked on this last year. Not sure if the code was commited to a PR anywhere.

**Comment by FrancoisPgm** [comment]
Yes I began exploring some solutions, I have a script to get the list of pull requests and contributers and update it, but it is not commited anywhere. It is not near ready for a PR yet. I'd be happy to share it if someone wants to continue working on it. I also might try to find time to continue it myself, but with the end of the PhD getting closer it's getting harder to reserve time for side projects.

**Comment by Remi-Gau** [comment]
chatted with @man-shu 
other things to track:
- what modules, functons, classes are most often used in nilearn

**Comment by Remi-Gau** [comment]
@man-shu 
reused some of your code and started created a mini-package to actually perform this type of 'audit' for any package

**Comment by Remi-Gau** [comment]
OK I moved the repo there:

https://github.com/nilearn/poia

@man-shu when you got a minute can you check you can run the notebook that's in there?

**Comment by Remi-Gau** [comment]
this may give us a figure for the OHBM poster to show how people use the package

**Comment by man-shu** [comment]
> @man-shu when you got a minute can you check you can run the notebook that's in there?

Yep works locally! But the CI is failing...

**Comment by Remi-Gau** [comment]
> > [@man-shu](https://github.com/man-shu) when you got a minute can you check you can run the notebook that's in there?
> 
> Yep works locally! But the CI is failing...

yeah need to fix that

I'd like to make sure the notebook is viewable in github pages directly rather than having to save the figures in the repo

**Comment by Remi-Gau** [comment]
OK the whole thing should now be browsable.

https://nilearn.github.io/poia/notebooks/poia.html

Important details: when deploying marimo notebooks files accesible on disk have to be accessed via request
